app/create_file.py

```
@@ -1 +1,127 @@
-# write your code here
+import sys
+import os
+from datetime import datetime
+from typing import LiteralString
+
+
+class MissingArgumentError(Exception):
+
+    def __init__(self,
+                 flag: str = "",
+                 message: str = "A required argument is missing."
+                 ) -> None:
+        self.message = message
+        self.flag = flag
+        super().__init__(self.message)
+
+    def __str__(self) -> str:
+        message_map = {
+            "-d": "A directory name is required.",
+            "-f": "A file name is required."
+        }
+        if self.flag in message_map:
+            return f"{self.message} \n\n{message_map[self.flag]}"
+        return self.message
+
+
+class MissingFlagError(MissingArgumentError):
+
+    def __init__(
+            self,
+            message: str = "A required flag is missing."
+    ) -> None:
+        super().__init__(message)
+
+    def __str__(self) -> str:
+        return (
+            f"{self.message} \n\n"
+            f"You must provide at least a -d flag "
+            f"and a directory name to create a directory. \n"
+            f"Usage: python {sys.argv[0]} -d <directory_name> \n"
+            f"Usage for recursive directories: "
+            f"python {sys.argv[0]} "
+            f"-d <directory_name> <directory_name> \n\n"
+            f"You must provide at least a -f flag "
+            f"and a file name to create a file. \n"
+            f"Usage: python {sys.argv[0]} -f <file_name> \n\n"
+            f"Usage for recursive files: "
+            f"python {sys.argv[0]} -d <directory_name> -f <file_name> \n"
+        )
+
+
+def make_path(parts: list) -> LiteralString | str | bytes:
+
+    if len(parts) > 0:
+        return os.path.join(*parts)
+    else:
+        return ""
```

`make_path` returns an empty string when `parts` is empty. Returning `""` allows downstream calls (e.g., `create_directories(make_path(...))`) to receive an invalid path and masks missing-argument conditions. Validate `parts` is non-empty instead of returning `""` and raise a clear error (for example `MissingArgumentError("-d")`) so callers don't attempt to create or use an invalid path. This is important because the script must "work when only -d, only -f, or both -d and -f are provided, handling each mode correctly."

---

Sorry - but make_path is only in use, if not an Argument- or Flag-Error before. Where is the problem?